### PR TITLE
Feature/internationalize date formatting of charts

### LINKED ISSUE #2900: Internationalize date formatting of charts
### Goal
- Make date format for all charts depends on the current language

joxi_screenshot_1505206783976
joxi_screenshot_1505207313774

### PR DESCRIPTION
**Pull request by deligence-dharmendra**
Closes  #2900

## Changes

- change in format date helper file
- add  getLocaleDateFormat method

## Developer checklist
This checklist is to be completed by the PR developer:

- [ ] Alternative solutions were compared/discussed before writing code
    - [ ] trade-offs with this solution are considered acceptable
- [ ] Code in this PR adheres to the [project styleguide](CONTRIBUTING.md)
- [ ] This pull request does not decrease project test coverage
- [ ] If the code changes existing database collection(s), migration has been written
- [ ] If UI texts are added or changed, all texts are internationalized

## Reviewer checklist
Reviewed by: @marla-singer @matleppa 

This list is to be completed by the pull request reviewer:
- [ ] Code works as described/expected
- [ ] Code seems to be error free
    - [ ] no browser console errors visible
    - [ ] no server console errors visible
    - [ ] passes CI build
- [ ] Code is written in a way that promotes maintainability
    - [ ] easy to understand
    - [ ] well organized
    - [ ] follows project coding standards and conventions
    - [ ] well documented

### CODE REVIEW

File: apinf_packages/dashboard/client/charts/median_response_time/median_response_time.js [hunk 1]
```diff
@@ -69,7 +72,7 @@ Template.medianResponseTime.onRendered(function () {
 
     // Get dates
     const labels = chartData.map(dataset => {
-      return moment(dataset.date).format('MM/DD');
+      return moment(dataset.date).format(getLocaleDateFormat());
```
Comment (marla-singer) [inline apinf_packages/dashboard/client/charts/median_response_time/median_response_time.js]: Not good enough. Get the locale format **outside of loop**

File: apinf_packages/dashboard/client/charts/requests_number/requests_number.js [hunk 2]
```diff
@@ -94,7 +97,7 @@ Template.requestTimeline.onRendered(function () {
 
     // Initialization
     const labels = selectedPathData.dates.map(date => {
-      return moment(date).format('MM/DD');
+      return moment(date).format(getLocaleDateFormat());
```
Comment (marla-singer) [inline apinf_packages/dashboard/client/charts/requests_number/requests_number.js]: Get the locale format **outside of loop**

File: apinf_packages/dashboard/client/charts/requests_over_time/requests_over_time.js [hunk 3]
```diff
@@ -69,7 +72,7 @@ Template.requestsOverTime.onRendered(function () {
 
     // Get dates
     const labels = chartData.map(dataset => {
-      return moment(dataset.date).format('MM/DD');
+      return moment(dataset.date).format(getLocaleDateFormat());
```
Comment (marla-singer) [inline apinf_packages/dashboard/client/charts/requests_over_time/requests_over_time.js]: Get the locale format **outside of loop**

File: apinf_packages/dashboard/client/charts/response_time/response_time.js [hunk 4]
```diff
@@ -89,7 +92,7 @@ Template.responseTimeTimeline.onRendered(function () {
 
     // Create labels value
     const labels = selectedPathData.dates.map(date => {
-      return moment(date).format('MM/DD');
+      return moment(date).format(getLocaleDateFormat());
```
Comment (marla-singer) [inline apinf_packages/dashboard/client/charts/response_time/response_time.js]: Get the locale format **outside of loop**

File: apinf_packages/dashboard/client/charts/unique_users/unique_users.js [hunk 5]
```diff
@@ -69,7 +72,7 @@ Template.uniqueUsersOverTime.onRendered(function () {
 
     // Get dates
     const labels = chartData.map(dataset => {
-      return moment(dataset.date).format('MM/DD');
+      return moment(dataset.date).format(getLocaleDateFormat());
```
Comment (marla-singer) [inline apinf_packages/dashboard/client/charts/unique_users/unique_users.js]: Get the locale format **outside of loop**

File: apinf_packages/core/helper_functions/format_date.js [hunk 6]
```diff
@@ -16,9 +16,24 @@ import 'moment/min/locales.min';
...
+export function formatDate (date, dateFormat = 'LL') {
   // Get current language
   const language = TAPi18n.getLanguage();
   // Return formatted date
   return moment(date).locale(language).format(dateFormat);
 }
+
+export function getLocaleDateFormat () {
+  // Get locale
+  const locale = window.navigator.userLanguage || window.navigator.language;
```
Comment (marla-singer) [inline apinf_packages/core/helper_functions/format_date.js]: Using another command to cover all user's preferred language:
```js
const locale = navigator.languages ? navigator.languages[0] : (navigator.language || navigator.userLanguage)
```

Comment (marla-singer) [reply apinf_packages/core/helper_functions/format_date.js]: And please squash all commits to one

File: apinf_packages/core/helper_functions/format_date.js [hunk 7]
```diff
@@ -16,9 +16,25 @@ import 'moment/min/locales.min';
...
+  const locale = navigator.languages ? navigator.languages[0] :
+    (navigator.language || navigator.userLanguage);
+  moment.locale(locale);
+
+  // Get locale data
+  const localeData = moment.localeData();
+  let format = localeData.longDateFormat('L');
+
+  // Remove year part
+  format = format.replace(/.YYYY/, '');
```
Comment (matleppa) [inline apinf_packages/core/helper_functions/format_date.js]: Remove only the year, not the dot before year.

Comment (marla-singer) [reply apinf_packages/core/helper_functions/format_date.js]: @matleppa For what? The label looks "07.02" and I don't understand why do we need save the last dot?

Comment (matleppa) [reply apinf_packages/core/helper_functions/format_date.js]: The label should look "07.02", because the last dot is part of the date in this form.

Comment (marla-singer) [reply apinf_packages/core/helper_functions/format_date.js]: Can't agree we use 👀  let's ask other devs about this?

Comment (marla-singer) [reply apinf_packages/core/helper_functions/format_date.js]: Honestly, I don't know about Finnish date format, but for me the last dot looks sooo strange in the Date

## Current realization
joxi_screenshot_1518080287117


## Your suggestions
joxi_screenshot_1518080241345

Comment (matleppa) [reply apinf_packages/core/helper_functions/format_date.js]: Isn't this matter exactly about formatting the dates according to selected language?
As you said, that you are not familiar with Finnish format of dates, so I can enlighten you.
The format in this case is "07.02.".

So let's format dates according to selected language.

Comment (marla-singer) [reply apinf_packages/core/helper_functions/format_date.js]: @matleppa Actually, the task is about User locale, not Selected language

Comment (matleppa) [reply apinf_packages/core/helper_functions/format_date.js]: After re-reading the issue, I see, that it indeed is about the User locale.
However, the question remains: When my locale is Finland, which date format should I see?

Comment (marla-singer) [reply apinf_packages/core/helper_functions/format_date.js]: > However, the question remains: When my locale is Finland, which date format should I see?

I have no idea
ping @anarva @preriasusi

Comment (preriasusi) [reply apinf_packages/core/helper_functions/format_date.js]: OK, so data format should come from computer locale. Not from apinf.io page language selection. @marla-singer has a point. Is it possible to get computer locale? If not, then we do a mapping from language to locale. It's not perfect solution, but in many cases looks great. We can live with it.

Comment (marla-singer) [reply apinf_packages/core/helper_functions/format_date.js]: ping @matleppa 
The PR can't be merged w/out your approve

Comment (preriasusi) [reply apinf_packages/core/helper_functions/format_date.js]: Keep the dot. In Finnish last dot is an indicator of the date. There's now dates without a fot after the month.

Comment (deligence-dharmendra) [reply apinf_packages/core/helper_functions/format_date.js]: @preriasusi, That means we go with keep the dot with finish date.

Comment (preriasusi) [reply apinf_packages/core/helper_functions/format_date.js]: First if you get a date format from a i18n component/library, it is not a good ide ato start clipping it. And also very bad idea to do now some logic according to the language. Isn't there a format for the date without a year that you could use without doing any mods on it?

Comment (marla-singer) [reply apinf_packages/core/helper_functions/format_date.js]: @preriasusi Nope. We use moment.js and it doesn't provide locale date w/out year

Comment (preriasusi) [reply apinf_packages/core/helper_functions/format_date.js]: @marla-singer Ok, it seems that it doesn't. So, ok stripping the year away is in this case accettable. But touching the dot in Finnish date is very, very wrong thing to do.

**Comment by marla-singer** [comment]
@matleppa Add case for Finnish locale. Please check and let's merge it

File: apinf_packages/core/helper_functions/format_date.js [hunk 8]
```diff
@@ -27,14 +27,21 @@ export function getLocaleDateFormat () {
...
   moment.locale(locale);
 
   // Get locale data
   const localeData = moment.localeData();
+
   let format = localeData.longDateFormat('L');
 
-  // Remove year part
-  format = format.replace(/.YYYY/, '');
+  if (locale === 'fi') {
```
Comment (matleppa) [inline apinf_packages/core/helper_functions/format_date.js]: Seems not to work.
Should the comparison be 
locale === 'fi-FI'

Comment (marla-singer) [reply apinf_packages/core/helper_functions/format_date.js]: I remade it by using regex. Please check again. If everything is fine then not merge it yet, I wanna do `rebase`

**Pull request merged**